### PR TITLE
Firebase integration with Crashlytics. Error handling, logging, & reporting

### PR DESCRIPTION
**Pull request by m-rosario**
Sets up the foundation for integrating Firebase into the application. The first Firebase service being used is Crashlytics, which will allow for error reporting in release builds.

Additionally, this commits creates the basic infrastructure for error handling by offering several fallback handlers, logging, and reporting. A custom ErrorWidget, Dialogs class, and a retry function are also included as utilities that can be leveraged by developers.

### CODE REVIEW

File: lib/utilities/errors/crash_handler.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  bool _isCollectionEnabled = false;
+
+  CrashHandler(bool isReleaseMode, bool isCollectionEnabled) {
+    _isReleaseMode = isReleaseMode;
+    _isCollectionEnabled = isCollectionEnabled;
+    _initialize();
+  }
+
+  void _initialize() async {
+    await Firebase.initializeApp();
```
Comment (ashlilili) [inline lib/utilities/errors/crash_handler.dart:24]: I pulled from this branch but failed at this step with web, fixed by changing it to `Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);`.

**Review by ashlilili — APPROVED** [review]
Super cool stuff! Thanks for adding the firebase integration & error handling logic. I'm excited to use crashHandler in the future, especially the retrying & snackbar logic like the demo we saw!

File: lib/utilities/errors/crash_handler.dart (hunk 1 above)
Comment (m-rosario) [reply lib/utilities/errors/crash_handler.dart]: Ah very nice catch! I just add a commit to fix this. Thanks!

**Review by ashlilili — APPROVED** [review]
LGTM. Thanks for updating github actions!

**Pull request merged**